src/plugin/any23/src/java/org/apache/nutch/any23/Any23ParseFilter.java

```diff
@@ -0,0 +1,165 @@
+import org.apache.any23.writer.TripleHandlerException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.nutch.metadata.Metadata;
+import org.apache.nutch.parse.*;
+import org.apache.nutch.protocol.Content;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.DocumentFragment;
+
+/**
+ * <p>This implementation of {@link org.apache.nutch.parse.HtmlParseFilter}
+ * uses the <a href="http://any23.apache.org">Apache Any23</a> library
+ * for parsing and extracting structured data in RDF format from a
+ * variety of Web documents. Currently it supports the following
+ * input formats:</p>
+ * <ol><li>RDF/XML, Turtle, Notation 3</li>
+ * <li>RDFa with RDFa1.1 prefix mechanism</li>
+ * <li>Microformats: Adr, Geo, hCalendar, hCard, hListing, hResume, hReview,
+ * License, XFN and Species</li>
+ * <li>HTML5 Microdata: (such as Schema.org)</li>
+ * <li>CSV: Comma Separated Values with separator autodetection.</li></ol>.
+ * <p>In this implementation triples are written as Notation3 e.g.
+ * <code><http://www.bbc.co.uk/news/scotland/> <http://iptc.org/std/rNews/2011-10-07#datePublished> "2014/03/31 13:53:03"@en-gb .</code>
+ * and triples are identified within output triple streams by the presence of '\n'.
+ * The presence of the '\n' is a characteristic specific to N3 serialization in Any23.
+ * In order to use another/other writers implementing the
+ * <a href="http://any23.apache.org/apidocs/index.html?org/apache/any23/writer/TripleHandler.html">TripleHandler</a>
+ * interface, we will most likely need to identify an alternative data characteristic
+ * which we can use to split triples streams.</p>
+ * <p>
+ *
+ */
+public class Any23ParseFilter implements HtmlParseFilter {
+
+  /** Logging instance */
+  public static final Logger LOG = LoggerFactory.getLogger(Any23ParseFilter.class);
+
+  private Configuration conf = null;
+
+  /** Constant identifier used as a Key for writing and reading
+   * triples to and from the metadata Map field.
+   */
+  public final static String ANY23_TRIPLES = "Any23-Triples";
+
+  private static class Any23Parser {
+
+    Set<String> triples = null;
+
+    Any23Parser(String url, String htmlContent) throws TripleHandlerException {
+      triples = new TreeSet<String>();
+      try {
+        parse(url, htmlContent);
+      } catch (URISyntaxException e) {
+        throw new RuntimeException(e.getReason());
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+
+    /**
+     * Maintains a {@link java.util.Set} containing the triples
+     * @return a {@link java.util.Set} of triples.
+     */
+    public Set<String> getTriples() {
+      return triples;
+    }
+
+    private void parse(String url, String htmlContent) throws URISyntaxException, IOException, TripleHandlerException {
+      Any23 any23 = new Any23();
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      TripleHandler tHandler = new NTriplesWriter(baos);
```

Comment:
Please write the triples as Turtle, it is easier to read and hence debug if this breaks in the future.

Reply:
Ignore the comment.